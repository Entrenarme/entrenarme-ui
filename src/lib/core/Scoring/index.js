// @flow
import * as React from 'react';
import styled from 'styled-components';
import times from 'lodash/times';

import { RESOURCES_URL } from '../../helpers/config';
import colors from '../../helpers/colors';
import { mainFont } from '../../helpers/fonts';

type Props = {
  children: React.Node,
  score?: number,
};

const MainContainer = styled.div`
  margin-bottom: 0.625rem;
`;

const ScoreItem = styled.div`
  width: 0.625rem;
  height: 0.625rem;
  mask: url(${`${RESOURCES_URL}static/images/general/hex_icon_v2.svg`});
  background-color: ${props =>
    props.mark ? colors.secondary.default : colors.gray.default};

  @media only screen and (max-width: 767px) {
    width: 0.5rem;
    height: 0.5rem;
  }
`;

const ScoreTitle = styled.div`
  font-family: ${mainFont};
  color: ${colors.text};
  text-transform: uppercase;
  font-weight: 600;
  margin-bottom: 0.3125rem;
  font-size: 0.75rem;

  @media only screen and (max-width: 767px) {
    font-size: 0.625rem;
  }
`;

const ScoreContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(5, 0.625rem);
  grid-gap: 0.4375rem;

  @media only screen and (max-width: 767px) {
    grid-template-columns: repeat(5, 0.5rem);
    grid-gap: 0.3125rem;
  }
`;

const Scoring = ({ children, score, ...rest }: Props) => {
  const renderScore = score => {
    const iterableArray = times(5, String);

    return iterableArray.map((item, index) => (
      <ScoreItem key={item} mark={score > index} />
    ));
  };

  return (
    <MainContainer>
      <ScoreTitle>{children}</ScoreTitle>
      <ScoreContainer>{renderScore(score)}</ScoreContainer>
    </MainContainer>
  );
};

Scoring.defaultProps = {
  score: 0,
};

export default Scoring;
