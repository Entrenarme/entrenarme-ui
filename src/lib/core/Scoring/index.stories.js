// @flow
import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import Scoring from './index';

const ElementDecoration = storyFn => (
  <div style={{ padding: '3rem' }}>{storyFn()}</div>
);

storiesOf('Scoring', module)
  .addDecorator(ElementDecoration)
  .add(
    'default',
    withInfo('Scoring')(() => <Scoring>Entrenamiento personal</Scoring>),
  );

storiesOf('Scoring', module)
  .addDecorator(ElementDecoration)
  .add(
    'score = 3',
    withInfo('Scoring')(() => (
      <Scoring score={3}>Entrenamiento personal</Scoring>
    )),
  );
