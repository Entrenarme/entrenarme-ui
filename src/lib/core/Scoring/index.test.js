import * as React from 'react';
import { render } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';

import Scoring from './index';

test('renders "Scoring" with no props', () => {
  const { container } = render(<Scoring>Paragraph</Scoring>);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders "Scoring" with score of 3', () => {
  const { container } = render(<Scoring score={3}>Paragraph</Scoring>);
  expect(container.firstChild).toMatchSnapshot();
});
