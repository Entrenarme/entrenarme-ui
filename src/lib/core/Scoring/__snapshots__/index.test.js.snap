// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders "Scoring" with no props 1`] = `
<div
  class="sc-bdVaJa jnyass"
>
  <div
    class="sc-htpNat fbLUHU"
  >
    Paragraph
  </div>
  <div
    class="sc-bxivhb hsGGdc"
  >
    <div
      class="sc-bwzfXH gPYwax"
    />
    <div
      class="sc-bwzfXH gPYwax"
    />
    <div
      class="sc-bwzfXH gPYwax"
    />
    <div
      class="sc-bwzfXH gPYwax"
    />
    <div
      class="sc-bwzfXH gPYwax"
    />
  </div>
</div>
`;

exports[`renders "Scoring" with score of 3 1`] = `
<div
  class="sc-bdVaJa jnyass"
>
  <div
    class="sc-htpNat fbLUHU"
  >
    Paragraph
  </div>
  <div
    class="sc-bxivhb hsGGdc"
  >
    <div
      class="sc-bwzfXH iTzJCO"
    />
    <div
      class="sc-bwzfXH iTzJCO"
    />
    <div
      class="sc-bwzfXH iTzJCO"
    />
    <div
      class="sc-bwzfXH gPYwax"
    />
    <div
      class="sc-bwzfXH gPYwax"
    />
  </div>
</div>
`;
